vendor/src/github.com/docker/libnetwork/resolver_unix.go

```diff
@@ -0,0 +1,77 @@
+// +build !windows
+
+package libnetwork
+
+import (
+	"fmt"
+	"net"
+	"os"
+	"os/exec"
+	"runtime"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/docker/docker/pkg/reexec"
+	"github.com/docker/libnetwork/iptables"
+	"github.com/vishvananda/netns"
+)
+
+func init() {
+	reexec.Register("setup-resolver", reexecSetupResolver)
```

Comment:
@icecrime The root cause of the issue is the way golang deals with OS thread creation (more details in #1113). Handling it in the context of another process is the cleanest way to avoid this issue.